### PR TITLE
[WB-51] Add support for non-strict Event wrappping during Event conve…

### PR DESCRIPTION
**Pull request by austinkelleher**
…rtData.

### CODE REVIEW

File: models/events/webhook/github/GithubCommit.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module.exports = require('../../../Model').extend({
```
Comment (austinkelleher) [inline models/events/webhook/github/GithubCommit.js]: Bonus change

File: models/events/webhook/github/GithubPush.js [hunk 2]
```diff
@@ -1,5 +1,21 @@
+const GithubCommit = require('./GithubCommit')
```
Comment (austinkelleher) [inline models/events/webhook/github/GithubPush.js]: Bonus change

File: models/events/webhook/github/GithubCommit.js (hunk 1 above)
Comment (Druotic) [inline models/events/webhook/github/GithubCommit.js]: Shouldn't this have a typeName and an entry in EventType, or is that future work (maybe that's what you meant by 'bonus change?')?

File: models/util/convert-data-mixin.js [hunk 3]
```diff
@@ -24,7 +24,12 @@ module.exports = {
         }
       }
 
-      data = Type.data.wrap(data, errors)
+      if (Type.data.strict === false) {
+        data = Type.data.wrap(data)
```
Comment (Druotic) [inline models/util/convert-data-mixin.js]: Won't this completely hide errors? Even if we're trying to be non-strict for future GH changes, we should probably at least log something to let us know it's happening.

File: queue/util/message-parser.js [hunk 4]
```diff
@@ -18,5 +18,15 @@ exports.encode = function (message) {
 */
 exports.decode = function (message) {
   message = msgpack.decode(message)
-  return Event.wrap(message)
+
+  // Always report errors for Event model wrapping. These properties should
+  // always exist even if the underlying data in the Event is corrupt.
+  let errors = []
```
Comment (charlieduong94) [inline queue/util/message-parser.js]: I'm down with adding `prefer-const`.

**Comment by austinkelleher** [comment]
[Add support for non-strict Event wrapping](https://trello.com/c/zzKcIr44/51-add-support-for-non-strict-event-wrapping)

**Pull request merged**